### DISCUSSION TITLE
Add data filtering through interceptors in the row

### DISCUSSION THREAD

**Discussion by jorgeijni** [serenity-is/Serenity#6653] [Help]
Hello, I have implemented a data filter by creating interceptors in the project so that when the row is marked as filtered by owner, it filters only the data of the user who has registered the record. This is working 100% but when they use the search option in the grid it always returns all the values ​​instead of showing only the search value.

I attach some screenshots of my code of how it is implemented
![Screenshot 2023-01-12 111947](https://user-images.githubusercontent.com/1426903/212136139-20eb3ad3-31d5-43be-9a44-4bc1866fdb55.png)
![Screenshot 2023-01-12 112144](https://user-images.githubusercontent.com/1426903/212136140-5ac665df-6811-4f0a-b27a-e1782efa1e9f.png)
![Screenshot 2023-01-12 112225](https://user-images.githubusercontent.com/1426903/212136141-e5a29c63-96dc-49db-9cf7-d46b3be5599b.png)

Thank you very much in advance for your help.

**Comment by wezmag** [comment]
I've never use interceptors before, but maybe you want to try the [Serenity Service Behaviors](https://serenity.is/docs/tutorials/multi_tenancy/using_serenity_service_behaviors).

**Reply by jorgeijni** [reply]
Thank you I will check and thanks for the clarification since in reality they are Behavior that I am using.
